src/scripts/main.js

```diff
@@ -1,3 +1,305 @@
 'use strict';
 
-// write your code here
+const scoreDisplay = document.querySelector('.game-score');
+const button = document.querySelector('.start');
+const cells = document.querySelectorAll('.field-cell');
+const messages = document.querySelectorAll('.message');
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+
+const gameSize = 4;
+let score = 0;
+const winScore = 2048;
+
+const gameGrid = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function getRandomNumber() {
+  return Math.floor(Math.random() * 4);
+}
+
+button.addEventListener('click', (e) => {
+  if (button.classList.contains('start')) {
+    startGame();
+    button.className = 'button restart';
+    button.textContent = 'Restart';
+
+    messages.forEach(message => {
+      message.classList.add('hidden');
+    });
+  } else if (button.classList.contains('restart')) {
+    loseMessage.classList.add('hidden');
+    winMessage.classList.add('hidden');
+
+    cells.forEach(cell => {
+      cell.textContent = '';
+      cell.className = 'field-cell';
+    });
+    resetGame();
+    startGame();
+    score = 0;
+    scoreDisplay.textContent = score;
+  }
+});
+
+document.addEventListener('keydown', e => {
+  e.preventDefault();
+
+  switch (e.key) {
+    case 'ArrowUp':
+      moveUp();
+      break;
+    case 'ArrowDown':
+      moveDown();
+      break;
+    case 'ArrowLeft':
+      moveLeft();
+      break;
+    case 'ArrowRight':
+      moveRight();
+      break;
+    default:
+      break;
+  }
+});
+
+function startGame() {
+  renderGameBoard();
+
+  for (let i = 0; i < 2; i++) {
+    setRandomCell();
+  }
+}
+
+function renderGameBoard() {
+  let maxNumber = 0;
+
+  for (let row = 0; row < gameSize; row++) {
+    for (let col = 0; col < gameSize; col++) {
+      const cellValue = gameGrid[row][col];
+      const cell = cells[row * gameSize + col];
+
+      cell.innerHTML = '';
+
+      if (cellValue !== 0) {
+        cell.classList.add(`cell-${cellValue}`);
+
+        if (cellValue > maxNumber) {
+          maxNumber = cellValue;
+        }
+      } else {
+        cell.classList.remove(`cell-${winScore}`);
+      }
+
+      if (cellValue !== 0) {
+        const cellText = document.createElement('span');
+
+        cellText.textContent = cellValue;
+        cell.appendChild(cellText);
+      }
+    }
+  }
+  score = maxNumber;
```

Comment:
The score should be calculated based on the sum of all merged cells, not the max cell value. Update the score calculation accordingly.